tutorials/dataframe/df037_TTreeEventMatching.py

```diff
@@ -0,0 +1,154 @@
+        with ROOT.TFile(self.aux_file_2, "RECREATE") as f:
+            aux_tree_2 = ROOT.TTree(self.aux_tree_name_2, self.aux_tree_name_2)
+            idx = numpy.array([0], dtype=int)
+            z = numpy.array([0], dtype=int)
+            aux_tree_2.Branch("idx", idx, "idx/I")
+            aux_tree_2.Branch("z", z, "z/I")
+
+            idx[0] = 1
+            z[0] = 6
+            aux_tree_2.Fill()
+            idx[0] = 3
+            z[0] = 7
+            aux_tree_2.Fill()
+
+            aux_tree_2.Write()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *_):
+        os.remove(self.main_file)
+        os.remove(self.aux_file_1)
+        os.remove(self.aux_file_2)
+
+
+def df037_TTreeEventMatching(dataset: DatasetContext):
+    # The input dataset has one main TTree and two auxiliary. The 'idx' branch
+    # is used as the index to match events between the trees.
+    # - The main tree has 3 entries, with 'idx' values(1, 2, 3).
+    # - The first auxiliary tree has 2 entries, with 'idx' values(1, 2).
+    # - The second auxiliary tree has 2 entries, with 'idx' values(1, 3).
+    # The two auxiliary trees are concatenated horizontally with the main one.
+    main_chain = ROOT.TChain(dataset.main_tree_name)
+    main_chain.Add(dataset.main_file)
+
+    aux_chain_1 = ROOT.TChain(dataset.aux_tree_name_1)
+    aux_chain_1.Add(dataset.aux_file_1)
+    aux_chain_1.BuildIndex("idx")
+
+    aux_chain_2 = ROOT.TChain(dataset.aux_tree_name_2)
+    aux_chain_2.Add(dataset.aux_file_2)
+    aux_chain_2.BuildIndex("idx")
+
+    main_chain.AddFriend(aux_chain_1)
+    main_chain.AddFriend(aux_chain_2)
+
+    # Create an RDataFrame to process the input dataset. The DefaultFor and
+    # SkipMissingEntries functionalities can be used to decide what to do for
+    # the events that do not match entirely according to the index column 'idx'
+    df = ROOT.RDataFrame(main_chain)
+
+    aux_tree_1_colidx = dataset.aux_tree_name_1 + ".idx"
+    aux_tree_1_coly = dataset.aux_tree_name_1 + ".y"
+    aux_tree_2_colidx = dataset.aux_tree_name_2 + ".idx"
+    aux_tree_2_colz = dataset.aux_tree_name_2 + ".z"
+
+    # Example 1: provide default values for all columns in case there was no
+    # match
+    display_1 = (
+        df.DefaultFor(aux_tree_1_colidx, -1)
+        .DefaultFor(aux_tree_1_coly, -2)
+        .DefaultFor(aux_tree_2_colidx, -3)
+        .DefaultFor(aux_tree_2_colz, -4)
+        .Display(
+            ("idx", aux_tree_1_colidx, aux_tree_2_colidx, "x", aux_tree_1_coly, aux_tree_2_colz))
+    )
+
+    # Example 2: Skip the entire entry when there was no match for a column
+    # in the first auxiliary tree, but keep the entries when there is no match
+    # in the second auxiliary tree and provide a default value for those
+    display_2 = (
+        df.DefaultFor(aux_tree_2_colidx, -3)
+        .DefaultFor(aux_tree_2_colz, -4)
+        .SkipMissingEntries(aux_tree_1_coly)
```

Comment:
Do we have an example of the case of `SkipMissingEntries()` (i.e. or the spelling of the case to skip missing entries for any columns)?

Reply:
That spelling was just an initial exploration of a potential feature. After much debugging and code analysis (both on my own and with others), I decided to remove that for better stability and reduced maintenance burden. I believe we should start by shipping a solid set of the two main features requested by users, then in the future we can accommodate more.